engine/src/main/java/org/camunda/bpm/engine/impl/cmd/FetchExternalTasksCmd.java

```diff
@@ -74,12 +76,24 @@ public FetchExternalTasksCmd(String workerId, int maxResults, Map<String, TopicF
     for (ExternalTaskEntity entity : externalTasks) {
       
       TopicFetchInstruction fetchInstruction = fetchInstructions.get(entity.getTopicName());
-      entity.lock(workerId, fetchInstruction.getLockDuration());
-
-      LockedExternalTaskImpl resultTask = LockedExternalTaskImpl.fromEntity(entity, fetchInstruction.getVariablesToFetch(), fetchInstruction.isLocalVariables(),
-          fetchInstruction.isDeserializeVariables(), fetchInstruction.isIncludeExtensionProperties());
-
-      result.add(resultTask);
+      //retrieve latest execution to avoid concurrent modifications to the execution @https://jira.camunda.com/browse/CAM-10750
+      ExecutionEntity execution = commandContext
+    		  .getExecutionManager()
+    		  .findExecutionById(entity.getExecutionId());
```

Comment:
Please consolidate this with `ExternalTaskEntity#getExecution`. E.g. you could add another flavour of that method that does not do the `null` validation. You then also won't to call `setExecution` explicitly below.